src/crypto/rand/rand_baremetal.go

```diff
@@ -0,0 +1,38 @@
+//go:build stm32wle5
+// +build stm32wle5
+
+package rand
+
+import (
+	"machine"
+)
+
+func init() {
+
+	Reader = &reader{}
+	machine.InitRNG()
```

Comment:
I'm not a big fan of the naming for these 2 functions or the tedious `InitRNG` as a _requirement_.

It would be simpler to do lazy initialization on the first call to `ReadRNG`. This simplifies the requirements on any `machine` that implements it to a single function call, which I would propose be renamed `GetRNG`, since it is a fixed-length return variable unlike the various `Read` functions.

If the `machine` chooses it wants to add automatic `initRNG`, it can do so privately in its own `init`.

Reply:
Sounds good to me. It does make reading random numbers _slightly_ slower but that's probably negligible and IMHO outweighs the API simplicity.

No strong opinions on `GetRNG` vs `ReadRNG` from me. I picked `ReadRNG` because it makes it clear that it's not just a getter, but on the other hand you also have `getrandom` and `getentropy` syscalls (with a `get` prefix).

Reply:
> If the `machine` chooses it wants to add automatic `initRNG`, it can do so privately in its own `init`.

I don't think we should do that because it means the RNG is always initialized, even when unused (which might result in higher current consumption, etc).